File: chatapp.client/src/app/models/message-admin-view-model.ts
import { MessageMediaViewModel } from "./message-media-view-model";

export class MessageAdminViewModel {
    id: number = 0;
    senderUsername: string = '';
    senderId: number = 0;
    recipientUsername: string = '';
    recipientId: number = 0;
    content: string = '';
    media: MessageMediaViewModel | null = null;
    type: string = '';
    hasMedia: boolean = false;
    isSeen: boolean = false;
    isDeleted: boolean = false;
    parentMessageId: number = 0;
    createdAt: string = '';
    modifiedAt: string = '';
}

File: ChatApp.Server/Services/ViewModels/Messages/MessageAdminViewModel.cs
using ChatApp.Server.Services.ViewModels.Media;

namespace ChatApp.Server.Services.ViewModels.Messages
{
    public class MessageAdminViewModel
    {
        public int Id { get; set; }

        public string SenderUsername { get; set; }

        public int SenderId { get; set; }

        public string RecipientUsername { get; set; }

        public int RecipientId { get; set; }

        public string Content { get; set; } = "";

        public bool HasMedia { get; set; }

        public MediaViewModel? Media { get; set; }

        public bool IsSeen { get; set; }

        public bool IsDeleted { get; set; }

        public int? ParentMessageId { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime ModifiedAt { get; set; }
    }
}